### PR TITLE
dandi ls: Make json and json_pp formats output arrays; add json_lines for old json format

### LINKED ISSUE #652: dandi ls -f json_pp doesn't produce a valid json
we wrapping into a list. ATM `-f json` produces legit "json lines" having each record on a separate line.  But I think we better make it into a  explicit `json_lines` or alike.

`json_pp` takes those lines and renders them across multiple lines thus making overall illegit.  For both `json` and `json_pp` let's place all the records into a list and output that one

note: -f yaml seems to do the right thing and makes a list

### PR DESCRIPTION
**Pull request by jwodder**
Closes #652.

### CODE REVIEW

File: dandi/cli/formatter.py [hunk 1]
```diff
@@ -22,21 +22,57 @@ class JSONFormatter(Formatter):
...
+
+    def __call__(self, rec):
+        import json
+        from textwrap import indent
+
+        if self.first:
+            print(file=self.out)
+            self.first = False
+        else:
+            print(",", file=self.out)
```
Comment (yarikoptic) [inline dandi/cli/formatter.py:49]: would you be so kind to add a proper unittest which would operate over some very simple datastructures but with at least 2 items so we could ensure that we are formatting multi-item output correctly?

**Review by yarikoptic — CHANGES_REQUESTED** [review]
Looks good, but I am asking for some new extra test to finally ensure that we are doing it correctly now

File: dandi/cli/formatter.py (hunk 1 above)
Comment (jwodder) [reply dandi/cli/formatter.py:49]: Just for `JSONFormatter` or also for `JSONLinesFormatter`?

Comment (yarikoptic) [reply dandi/cli/formatter.py:49]: thanks for the tests, please add one for `JSONLinesFormatter` as well while at it -- would only help

Comment (jwodder) [reply dandi/cli/formatter.py:49]: Done.

**Comment by yarikoptic** [comment]
Looks good, thank you @jwodder !

**Pull request merged**